frontend/context.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+type contextKey int
+
+const (
+	// Keys for request-scoped data in http.Request contexts
+	contextKeyOriginalPath contextKey = iota
+	contextKeyBody
+	contextKeyLogger
+	contextKeyVersion
+	contextKeyCorrelationData
+	contextKeySystemData
+)
+
+func ContextWithOriginalPath(ctx context.Context, originalPath string) context.Context {
```

Comment:
This is adding lots of boilerplate (now we need to implement two functions for every new contextKey we add), but I understand we want to abstract out having to know which data type belongs to which contextKey.

I'm thinking we should try to enforce this somehow, maybe with tests, or by creating an interface that binds a key to is corresponding functions for using it, and then we only expose the interface and not these functions within.

Reply:
I have added now a simple error type that can be used when the value is not present in the context, instead of panicking (which we should avoid whenever possible).

I am not sure right now what would be the best to do about what you mention, as I am not sure if we need that type of extra binding using an interface.

One easy way to enforce that is to put all those keys and functions using the context in a new package and just export the functions we want (IMHO that would be the best option and could be done in a follow-up PR)

Reply:
I agree with abstracting out the keys further, and forcing the use of our functions to interact with the context.